test/Microsoft.Health.Dicom.Web.Tests.E2E/Common/DicomTagsManager.cs

```diff
@@ -114,8 +107,16 @@ public async Task<GetExtendedQueryTagEntry> UpdateExtendedQueryTagAsync(string t
     public async Task<bool> DeleteExtendedQueryTagAsync(string tagPath, CancellationToken cancellationToken = default)
     {
         EnsureArg.IsNotNull(tagPath, nameof(tagPath));
+        _tags.Remove(tagPath);
 
-        var response = await _dicomWebClient.DeleteExtendedQueryTagAsync(tagPath, cancellationToken);
-        return response.StatusCode == System.Net.HttpStatusCode.NoContent;
+        try
+        {
+            var response = await _dicomWebClient.DeleteExtendedQueryTagAsync(tagPath, cancellationToken);
+            return response.StatusCode == System.Net.HttpStatusCode.NoContent;
+        }
+        catch (DicomWebException dwe) when (dwe.StatusCode == System.Net.HttpStatusCode.NotFound)
+        {
+            return false;
```

Comment:
Why not return `true` here? This method is only being called as a cleanup utility, not to validate the deletion response.

Reply:
Delete will throw an exception if the status code is not success, so it should only return true when it really deleted an entry. But I think it could be useful for the return value to indicate whether deleted succeeded, even if it's not used in this particular scenario.